### PR TITLE
gas: check user supplied gas price against the reference gas price

### PR DESCRIPTION
**Pull request by bmwill**

### CODE REVIEW

File: crates/sui-core/src/transaction_input_checker.rs [hunk 1]
```diff
@@ -158,6 +158,21 @@ async fn check_gas(
             TransactionKind::Single(SingleTransactionKind::PaySui(t)) => t.amounts.iter().sum(),
             _ => 0,
         };
+
+        // Ensure that a supplied transaction has specified a sufficiently high gas price,
+        // otherwise reject the transaction
+        let reference_gas_price = store
```
Comment (lxfind) [inline crates/sui-core/src/transaction_input_checker.rs:164]: Unfortunately this doesn't work at epoch boundaries. Specifically, the reference price from the sui system state object isn't guaranteed to be consistent with the current epoch

**Review by lxfind — CHANGES_REQUESTED** [review]

File: crates/sui-core/src/transaction_input_checker.rs (hunk 1 above)
Comment (bmwill) [reply crates/sui-core/src/transaction_input_checker.rs:164]: the price shouldn't ever be updated until the change epoch transaction is executed so this should technically work. Although if you'd prefer handling this another way how would you suggest we do this, given we need something to land by end of day.

Comment (lxfind) [reply crates/sui-core/src/transaction_input_checker.rs:164]: But that transaction isn't guaranteed to be executed as the last transaction, right? The checkpoint executor could execute tx out of order (purely dependency driven)

**Comment by bmwill** [comment]
Closing as @dariorussi is implementing this in a better way

**Pull request closed without merging**